---
 startx.cpp |   15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

Index: xinit/startx.cpp
===================================================================
--- xinit.orig/startx.cpp	2009-12-06 23:12:44.000000000 +0100
+++ xinit/startx.cpp	2009-12-06 23:23:41.000000000 +0100
@@ -242,18 +242,17 @@
     removelist=
 
     XCOMM set up default Xauth info for this machine
-    case `uname` in
-    Linux*)
+
+    XCOMM check for GNU hostname
+    if hostname --version > /dev/null 2>&1; then
         if [ -z "`hostname --version 2>&1 | grep GNU`" ]; then
             hostname=`hostname -f`
-        else
-            hostname=`hostname`
         fi
-        ;;
-    *)
+    fi
+
+    if [ -z "$hostname" ]; then
         hostname=`hostname`
-        ;;
-    esac
+    fi
 
     authdisplay=${display:-:0}
 #if defined(HAS_COOKIE_MAKER) && defined(MK_COOKIE)
